mino/minogrpc/server.go

```diff
@@ -215,21 +215,60 @@ func (rpc RPC) Stream(ctx context.Context,
 					return
 				}
 				if err != nil {
-					err = xerrors.Errorf("failed to receive for client '%s': %v",
-						addr.String(), err)
-					fabric.Logger.Err(err).Send()
-					errs <- err
+					fillAndLog(errs, xerrors.Errorf("client '%s' failed to "+
+						"receive from RPC server: %v", addr.String(), err))
 					return
 				}
 
-				orchRecv.in <- msg
+				// check if this is an envelop that needs to be relayed
+				enveloppe := &Envelope{}
+				err = ptypes.UnmarshalAny(msg.Message, enveloppe)
+				if err == nil && len(enveloppe.To) != 0 {
+					err = relayMessage(enveloppe, orchSender)
+					if err != nil {
+						fillAndLog(errs, xerrors.Errorf("failed to relay "+
+							"message: %v", err))
+						return
+					}
+				} else {
+					anyEnvelope, err := ptypes.MarshalAny(&Envelope{
+						Message: msg.Message})
+					if err != nil {
+						fillAndLog(errs, encoding.NewAnyEncodingError(anyEnvelope, err))
+						return
+					}
+
+					orchRecv.in <- &OverlayMsg{Message: anyEnvelope}
```

Comment:
Do you really need to have both types of message ? Isn't an envelop simply an overlay message ?

Reply:
True, I was getting closer to this conclusion. I am just concerned that using only enveloppes will be insufficient for more advanced routing. What do you think?